server/tests/e2e/test_api_cache.py

```diff
@@ -0,0 +1,25 @@
+import os
+import unittest
+
+import requests
+
+
+@unittest.skipIf(os.getenv("CXG_API_BASE") is None, "CXG_API_BASE not defined")
+class TestAPICache(unittest.TestCase):
+    def setUp(self):
+        self.apiUrlBase = os.environ["CXG_API_BASE"]
+
+    def test_dataset_metadata(self):
+        response = requests.get("/".join([self.apiUrlBase, "dp/v1/collections"]))
+        response.raise_for_status()
+        collection_id = response.json()['collections'][0]['id']
+        response = requests.get("/".join([self.apiUrlBase, f"dp/v1/collections/{collection_id}"]))
+        response.raise_for_status()
+        dataset_id = response.json()['datasets'][0]['id']
+        response = requests.head('/'.join([self.apiUrlBase, 'cellxgene/e', f"{dataset_id}.cxg", 'api/v0.3/dataset-metadata']))
```

Comment:
nit: Since the above is being used to determine a valid dataset only for the test, would section off the code into setup/exercise/verify blocks using comments.